thirdeye/thirdeye-frontend/app/utils/yaml-tools.js

```diff
@@ -29,6 +28,20 @@ rules:
     type: DATA_SLA              # Alert if data is missing.
     params:
       sla: 3_DAYS               # Data is missing for 3 days since last availability
+
+# You can mention a cron to allow this detection to be run periodically
+# For example, the cron below would execute every 5 minutes.
+# cron: "0 0/5 * 1/1 * ? *"
+
+# Backfill detections
+# You can use the backfillStart parameter to denote how far back you want the anomalies
+# to be detected.
+# A valid entry is a timestamp value in millis.
+# 
+# - By default, (if not mentioned), the lookback is 30 days
+# - A value of 0 (zero) implies complete backfill till the start of data. As shown below.
+# backfillStart: 0 
```

Comment:
Can you shed some light on the need for a custom backfill configuration? Since this is a one time setting, I am not very confident of putting it in the detection config.

Reply:
Agreed. But I found this to be the easiest way of having this capability without a larger change. 

So, here is the use case: I uploaded a new dataset and I want to see the anomalies. I was unable to find a way to do this. Creating an alert would trigger detections for the past month but not before that. It is not configurable. And I couldn't find a way to just fire a detection from the UI.

This parameter allows the user to have the ability to use a default or choose a start time at the creation of the alert. I agree with you that creating an alert rule and running detection can be separated. This change sort of addresses the immediate requirement that allows users to upload their datasets and run detections on past data. Let me know what you think.

Reply:
Couple of issues/suggestions
1.  There is an api through which you can trigger a replay or backfill. You might want to consider exposing this api to the user. It is better to keep the alert creation independent of backfill. 30 days is hard-coded just to populate some metrics on the UI for users to play around.
2. If you haven't explored yet, there is a preview experience through which users can trigger detection on older data. However, this is adhoc and there is no option of saving the results.